flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/TransformParser.java

```diff
@@ -384,10 +481,73 @@ public static SqlSelect parseFilterExpression(String filterExpression) {
         StringBuilder statement = new StringBuilder();
         statement.append("SELECT * FROM ");
         statement.append(DEFAULT_TABLE);
-        if (!StringUtils.isNullOrWhitespaceOnly(filterExpression)) {
+        if (!isNullOrWhitespaceOnly(filterExpression)) {
             statement.append(" WHERE ");
             statement.append(filterExpression);
         }
         return parseSelect(statement.toString());
     }
+
+    public static SqlNode rewriteExpression(SqlNode sqlNode, Map<String, SqlNode> replaceMap) {
+        if (sqlNode instanceof SqlBasicCall) {
+            SqlBasicCall sqlBasicCall = (SqlBasicCall) sqlNode;
```

Comment:
Need to handle SqlCase.

Reply:
Replace `SqlCall` instead of `SqlBaseCall` to cover `SqlCase`.